### PR TITLE
#1717 adding negative 0 example

### LINKED ISSUE #1717: Add docs test for negative zero in FixedDecimal
The [`FixedDecimal` docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html) don't strictly comment on whether negative zero should be treated as a separate concept from (positive) zero.  

Even though I think mathematics would treat -0 as equivalent to 0 (similar 0.999... is equivalent to 1), IEEE 754 for floating point defines the sign bit separate from the mantissa and exponent, allowing for -0.

ICU4X's implementation allows for -0 as a result of the IEEE spec's affordance for it.  ~~However, this is not specified either in docs or examples.~~   Even though negative zero is discussed in the API docs under [`Signum`](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/signum/enum.Signum.html),  users may not realize that it implies the same for `FixedDecimal` unless they happen upon the `Signum` module doc.

We should add an docs example with a FixedDecimal of -0 to communicate in the main `FixedDecimal` APIs that it is supported.

### PR DESCRIPTION
**Pull request by andrewpollack**
Supplementing  `utils/fixed_decimal_src/decimal.rs` main example with comment and example demonstrating negative 0. 

Fixes #1717

### CODE REVIEW

File: utils/fixed_decimal/src/decimal.rs [hunk 1]
```diff
@@ -49,6 +49,14 @@ const_assert!(core::mem::size_of::<usize>() >= core::mem::size_of::<u16>());
 ///
 /// dec.multiply_pow10(-2);
 /// assert_eq!("2.50", dec.to_string());
+///
+/// // IEEE 754 for floating point defines the sign bit separate
```
Comment (sffc) [inline utils/fixed_decimal/src/decimal.rs:53]: Nit: Put this example in the docs for `from_f64` to demonstrate that we retain the sign of the `-0.0` literal.

**Review by sffc — CHANGES_REQUESTED** [review]

File: utils/fixed_decimal/src/decimal.rs (hunk 1 above)
Comment (andrewpollack) [reply utils/fixed_decimal/src/decimal.rs:53]: Created the example at https://github.com/unicode-org/icu4x/pull/1807/files#diff-5851ca6096651367eced534b6a8b82f133009b5c71dc0e5963bd3a8fdaa7da5bR960!

I wasn't sure if you meant to remove the original example from the top-layer, so both are part of this PR

**Review by echeran — APPROVED** [review]

File: utils/fixed_decimal/src/decimal.rs [hunk 2]
```diff
@@ -948,6 +956,14 @@ impl FixedDecimal {
     /// )
     /// .expect("Finite, integer-valued quantity");
     /// assert_eq!(decimal.write_to_string(), "12345678000");
+    ///
+    /// // IEEE 754 for floating point defines the sign bit separate
```
Comment (sffc) [inline utils/fixed_decimal/src/decimal.rs:960]: Nit: Make these actually separate examples

```suggestion
    /// assert_eq!(decimal.write_to_string(), "12345678000");
    /// ```
    ///
    /// Negative zero is supported:
    ///
    /// ```
    /// // IEEE 754 for floating point defines the sign bit separate
```

**Comment by andrewpollack** [comment]
I moved the negative zero example away from the top-level docs, and into two locations:
* `FixedDecimal::try_from_f64`
  * To demonstrate that we retain the sign of the -0.0 literal
* `FixedDecimal::negate`
  * I believe this would be a location people would go to find this information. Includes examples on how to set the negative zero value

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
Oops, I didn't notice that @echeran was requested for review. @echeran Feel free to leave additional comments.

**Comment by echeran** [comment]
Everything looks fine, thanks for the PR & merge.
